### PR TITLE
Implement default root-endpoint for new API

### LINKED ISSUE #2967: API: Flask - Implement new Default endpoint
Reimplementation of the default root-endpoint implemented on nodejs-based to python for the new API.

Principal roadmap:  https://github.com/wazuh/wazuh/issues/2640

Need to add default request to the API spec.

Default root-endpoint to reimplement:
* [x] GET /

### PR DESCRIPTION
**Pull request by davidjiglesias**
Hi team,

This PR closes #2967. Implemented controller for default endpoint.

Best regards,

David J. Iglesias

**Results for API request test:**
```
root@2c15633ee26d:/# curl "http://localhost:55000" -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ3YXp1aCIsImlhdCI6MTU1NDM3MTk0OCwiZXhwIjoxNTU0MzcyNTQ4LCJzdWIiOiJmb28ifQ.zaAZx1UmHfWAMkKdh57joJDL894xeSt70xf51dJFG-Q"
{
  "api_version": "4.0.0",
  "hostname": "2c15633ee26d",
  "license_name": "GPL 2.0",
  "license_url": "https://github.com/wazuh/wazuh/blob/master/LICENSE",
  "revision": 4000,
  "timestamp": "2019-04-04T10:00:08+0000",
  "title": "Wazuh API"
}
```
**Results for unit test:**
```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.4.0, py-1.8.0, pluggy-0.9.0
rootdir: /var/ossec/framework/python/bin
collected 1 item                                                               

test/test_default_controller.py .                                        [100%]

=============================== warnings summary ===============================
test/test_default_controller.py::test_default_info
  /var/ossec/framework/python/lib/python3.7/site-packages/yaml/constructor.py:126: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    if not isinstance(key, collections.Hashable):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================== 1 passed, 1 warnings in 1.78 seconds =====================
```

### CODE REVIEW

File: api/api/models/basic_info.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            'timestamp': str
+        }
+
+        self.attribute_map = {
+            'title': 'title',
+            'api_version': 'api_version',
+            'revision': 'revision',
+            'license_name': 'license_name',
+            'license_url': 'license_url',
+            'hostname': 'license_url',
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
            'hostname': 'hostname',
```

File: api/api/models/basic_info.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+    @property
+    def api_version(self) -> str:
+        """Gets the api_version of this BasicInfo.
+
+
+        :return: The api_version of this BasicInfo.
+        :rtype: str
+        """
+        return self.api_version
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
        return self._api_version
```

File: api/api/models/basic_info.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+    @property
+    def revision(self) -> int:
+        """Gets the revision of this BasicInfo.
+
+
+        :return: The revision of this BasicInfo.
+        :rtype: int
+        """
+        return self.revision
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
        return self._revision
```

File: api/api/models/basic_info.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    @property
+    def license_name(self) -> str:
+        """Gets the license_name of this BasicInfo.
+
+
+        :return: The license_name of this BasicInfo.
+        :rtype: str
+        """
+        return self.license_name
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
        return self._license_name
```

File: api/api/models/basic_info.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+    @property
+    def license_url(self) -> str:
+        """Gets the license_url of this BasicInfo.
+
+
+        :return: The license_url of this BasicInfo.
+        :rtype: str
+        """
+        return self.license_url
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
        return self._license_url
```

File: api/api/models/basic_info.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+    @license_url.setter
+    def license_url(self, license_url: str):
+        """Sets the license_url of this BasicInfo.
+
```
Comment (crd1985) [inline api/api/models/basic_info.py]: Remove blank line

File: api/api/models/basic_info.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+    @property
+    def hostname(self) -> str:
+        """Gets the hostname of this BasicInfo.
+
+
+        :return: The hostname of this BasicInfo.
+        :rtype: str
+        """
+        return self.hostname
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
        return self._hostname
```

File: api/api/models/basic_info.py [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+    @property
+    def timestamp(self) -> str:
+        """Gets the timestamp of this BasicInfo.
+
+
+        :return: The timestamp of this BasicInfo.
+        :rtype: str
+        """
+        return self.timestamp
```
Comment (crd1985) [inline api/api/models/basic_info.py]:
```suggestion
        return self._timestamp
```

File: api/api/spec/spec.yaml [hunk 9]
```diff
@@ -103,6 +104,30 @@ components:
...
+          format: int32
+        license_name:
+          type: string
+          description: API license name.
+        license_url:
+          type: string
+          description: API license url.
+        hostname:
+          type: string
+          description: Machine´s hostname.
```
Comment (crd1985) [inline api/api/spec/spec.yaml]:
```suggestion
          description: Server hostname.
```

File: api/api/controllers/default_controller.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        data = {
+            'title': info_data['info']['title'],
+            'api_version': info_data['info']['version'],
+            'revision': info_data['info']['x-revision'],
+            'license_name': info_data['info']['license']['name'],
+            'license_url': info_data['info']['license']['url'],
+            'hostname': socket.gethostname(),
+            'timestamp': timestamp
+        }
+
```
Comment (crd1985) [inline api/api/controllers/default_controller.py]:
```suggestion
response = BasicInfo.from_dict(data)
```

File: api/api/controllers/default_controller.py [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    return data, 200
```
Comment (crd1985) [inline api/api/controllers/default_controller.py]:
```suggestion
    return response, 200
```

**Comment by davidjiglesias** [comment]
Hi team,

I have included an unit test for `BasicInfo` model. Also added changes to `default_controller` unit test and the controller response.

Best regards,

David J. Iglesias

**API test for default root endpoint:**
```
root@335bd17bf695:/# curl "http://localhost:55000/"
{
  "api_version": "4.0.0",
  "hostname": "335bd17bf695",
  "license_name": "GPL 2.0",
  "license_url": "https://github.com/wazuh/wazuh/blob/master/LICENSE",
  "revision": 4000,
  "timestamp": "2019-04-12T14:08:19+0000",
  "title": "Wazuh API"
}
```
**Controller unit test:**
```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.4.0, py-1.8.0, pluggy-0.9.0
rootdir: /var/ossec/framework/python/bin
collected 1 item                                                               

test/test_default_controller.py .                                        [100%]

=============================== warnings summary ===============================
/var/ossec/framework/python/lib/python3.7/site-packages/jsonschema/compat.py:6
/var/ossec/framework/python/lib/python3.7/site-packages/jsonschema/compat.py:6
  /var/ossec/framework/python/lib/python3.7/site-packages/jsonschema/compat.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import MutableMapping, Sequence  # noqa

/var/ossec/framework/python/lib/python3.7/site-packages/yaml/constructor.py:126
  /var/ossec/framework/python/lib/python3.7/site-packages/yaml/constructor.py:126: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    if not isinstance(key, collections.Hashable):

/var/ossec/framework/python/lib/python3.7/site-packages/jinja2/runtime.py:318
  /var/ossec/framework/python/lib/python3.7/site-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================== 1 passed, 4 warnings in 2.34 seconds =====================
```

**BasicInfo model unit test**
```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.4.0, py-1.8.0, pluggy-0.9.0
rootdir: /var/ossec/framework/python/bin
collected 1 item                                                               

test/test_basic_info.py .                                                [100%]

=============================== warnings summary ===============================
/var/ossec/framework/python/lib/python3.7/site-packages/jsonschema/compat.py:6
/var/ossec/framework/python/lib/python3.7/site-packages/jsonschema/compat.py:6
  /var/ossec/framework/python/lib/python3.7/site-packages/jsonschema/compat.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import MutableMapping, Sequence  # noqa

/var/ossec/framework/python/lib/python3.7/site-packages/yaml/constructor.py:126
  /var/ossec/framework/python/lib/python3.7/site-packages/yaml/constructor.py:126: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    if not isinstance(key, collections.Hashable):

/var/ossec/framework/python/lib/python3.7/site-packages/jinja2/runtime.py:318
  /var/ossec/framework/python/lib/python3.7/site-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================== 1 passed, 4 warnings in 0.57 seconds =====================
```

File: api/api/controllers/default_controller.py [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+
+@exception_handler
+def default_info():
+    """Get basicinfo
+
+    Returns various basic information about the API
+    """
+    with open(os.path.join(api_path[0], 'spec', 'spec.yaml'), 'r') as stream:
+        info_data = yaml.load(stream)
```
Comment (crd1985) [inline api/api/controllers/default_controller.py]: Is there any way to take the spec directly loaded from connexion instead of reloading the file?

**Comment by davidjiglesias** [comment]
**Results for Integration test:**
```
=================================== test session starts ====================================
platform linux -- Python 3.7.2, pytest-4.4.1, py-1.8.0, pluggy-0.9.0
rootdir: /var/ossec/framework/python/bin
plugins: tavern-0.26.3
collected 1 item                                                                           

test/test_default_endpoints.tavern.yaml .                                            [100%]

===================================== warnings summary =====================================
/var/ossec/framework/python/lib/python3.7/site-packages/PyYAML-3.13-py3.7-linux-x86_64.egg/yaml/constructor.py:126
  /var/ossec/framework/python/lib/python3.7/site-packages/PyYAML-3.13-py3.7-linux-x86_64.egg/yaml/constructor.py:126: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working

test/test_default_endpoints.tavern.yaml::GET /
  /var/ossec/framework/python/lib/python3.7/site-packages/jwt/api_jwt.py:81: DeprecationWarning: It is strongly recommended that you pass in a value for the "algorithms" argument when calling decode(). This argument will be mandatory in a future version.
    DeprecationWarning

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== 1 passed, 2 warnings in 2.07 seconds ===========================
```

File: api/api/controllers/default_controller.py (hunk 12 above)
Comment (crd1985) [reply api/api/controllers/default_controller.py]: Ok, we didn't manage to find a better way, so let's go on with this approach.

**Pull request merged**